daphne/src/roles_test.rs

```diff
@@ -4,39 +4,117 @@
 use crate::{
     auth::BearerToken,
     constants::{MEDIA_TYPE_AGG_INIT_REQ, MEDIA_TYPE_AGG_SHARE_REQ, MEDIA_TYPE_COLLECT_REQ},
-    hpke::{HpkeDecrypter, HpkeReceiverConfig},
+    hpke::HpkeReceiverConfig,
     messages::{
         AggregateInitializeReq, AggregateResp, AggregateShareReq, CollectReq, HpkeCiphertext, Id,
         Interval, Nonce, Report, ReportShare, TransitionFailure, TransitionVar,
     },
     roles::{DapAggregator, DapHelper, DapLeader},
-    testing::{MockAggregator, HPKE_RECEIVER_CONFIG_LIST, LEADER_BEARER_TOKEN},
-    DapAbort, DapError, DapMeasurement, DapRequest, Prio3Config, VdafConfig,
+    testing::{
+        BucketInfo, MockAggregator, ReportStore, HPKE_RECEIVER_CONFIG_LIST, LEADER_BEARER_TOKEN,
+    },
+    DapAbort, DapMeasurement, DapRequest, Prio3Config, VdafConfig,
 };
 use assert_matches::assert_matches;
 use prio::codec::{Decode, Encode};
-use rand::prelude::*;
-use std::vec;
+use std::{ops::DerefMut, vec};
 
-#[tokio::test]
-async fn http_post_aggregate_unauthorized_request() {
-    let mut rng = thread_rng();
-    let helper = MockAggregator::new();
-    let task_id = helper.nominal_task_id();
-    let task_config = helper.get_task_config_for(task_id).unwrap();
+fn gen_agg_init_req(
+    agg: &MockAggregator,
+    report_shares: Vec<ReportShare>,
+    sender_auth: Option<BearerToken>,
+) -> DapRequest<BearerToken> {
+    let task_id = agg.nominal_task_id();
+    let task_config = agg.get_task_config_for(task_id).unwrap();
 
-    let mut req = DapRequest {
+    DapRequest {
         media_type: Some(MEDIA_TYPE_AGG_INIT_REQ),
         payload: AggregateInitializeReq {
             task_id: task_id.clone(),
-            agg_job_id: Id(rng.gen()),
+            agg_job_id: Id([1; 32]),
```

Comment:
This should be generated at random:
```suggestion
            agg_job_id: Id(rng.gen()),
```
You'll need `let mut rng = thread_rng()` above.

Reply:
You are right. Got carried away when writing the request generation functions.